### PR TITLE
Add test for issue-58951

### LINKED ISSUE #58951: ICE with `#![feature(existential_type)]` + closure
Encountered this when working on test cases for an RFC:
```rust
#![feature(existential_type)]

existential type A: Iterator;
fn def_a() -> A { 0..1 }
pub fn use_a() {
    def_a().map(|x| x);
}
```
cc @oli-obk

This seems related to other issues.

### PR DESCRIPTION
**Pull request by JohnTitor**
Closes #58951

r? @Centril

### CODE REVIEW

File: src/test/ui/existential_types/issue-58951.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// run-pass
```
Comment (Centril) [inline src/test/ui/existential_types/issue-58951.rs]:
```suggestion
// check-pass
```

**Review by Centril — COMMENTED** [review]
r=me rollup with suggestion applied

File: src/test/ui/existential_types/issue-58951.rs (hunk 1 above)
Comment (JohnTitor) [reply src/test/ui/existential_types/issue-58951.rs]: Done!

**Pull request merged**